### PR TITLE
Use new wast parser in wasm2js

### PR DESCRIPTION
**Pull request by tlively**
When generating assertions, traverse the `WASTScript` data structure rather than
interleaving assertion parsing with emitting.

### CODE REVIEW

**Comment by tlively** [comment]
* **#6606** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6606?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tlively and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/wasm2js/br.2asm.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ function asmFunc(imports) {
   
  }
  
+ function $0() {
+  
+ }
```
Comment (kripken) [inline test/wasm2js/br.2asm.js:23]: What is the reason for these changes?

Comment (tlively) [reply test/wasm2js/br.2asm.js:23]: I have no idea, and not for lack of trying to figure it out :/

Comment (kripken) [reply test/wasm2js/br.2asm.js:23]: Perhaps `wasm-reduce` can help here? It could compare the old and new outputs and find the smallest possible input that causes a difference.

Comment (tlively) [reply test/wasm2js/br.2asm.js:23]: Oh, I thought this was an extra empty function that was being inserted at the beginning, but it's just the names that are being changed. The addition of function `$0` in this diff is balanced by the removal of old function `$4`. The difference in function names is just a difference between the old and new text parsers when parsing unnamed functions.

**Review by kripken — APPROVED** [review]
I see, thanks.

The small changes do make reviewing a little hard but I didn't see anything obviously worrying, and we do execute those outputs in `check.py`, so I'm not worried. We also have the emscripten wasm2js tests on the bots that will give additional coverage.
